app/server/src/main/scala/org/bitcoins/server/BitcoinSRunner.scala

```diff
@@ -0,0 +1,95 @@
+package org.bitcoins.server
+
+import java.nio.file.{Path, Paths}
+
+import akka.actor.ActorSystem
+import com.typesafe.config.{Config, ConfigFactory}
+import org.bitcoins.core.config._
+import org.bitcoins.core.util.BitcoinSLogger
+import org.bitcoins.db.{AppConfig, ConfigOps}
+
+import scala.concurrent.ExecutionContext
+import scala.util.Properties
+
+trait BitcoinSRunner extends BitcoinSLogger {
+
+  protected def args: Array[String]
+
+  def actorSystemName: String
+
+  implicit lazy val system: ActorSystem = {
+    val system = ActorSystem(actorSystemName, baseConfig)
+    system.log.info("Akka logger started")
+    system
+  }
+  implicit lazy val ec: ExecutionContext = system.dispatcher
+
+  lazy val argsWithIndex: Vector[(String, Int)] = args.toVector.zipWithIndex
+
+  lazy val rpcPortOpt: Option[Int] = {
+    val portOpt = argsWithIndex.find(_._1.toLowerCase == "--rpcport")
+    portOpt.map {
+      case (_, idx) => args(idx + 1).toInt
+    }
+  }
+
+  lazy val forceChainWorkRecalc: Boolean =
+    args.exists(_.toLowerCase == "--force-recalc-chainwork")
+
+  private lazy val dataDirIndexOpt: Option[(String, Int)] = {
+    argsWithIndex.find(_._1.toLowerCase == "--datadir")
+  }
+
+  private lazy val datadirPathOpt: Option[Path] = dataDirIndexOpt match {
+    case None => None
+    case Some((_, dataDirIndex)) =>
+      val str = args(dataDirIndex + 1)
+      val usableStr = str.replace("~", Properties.userHome)
+      Some(Paths.get(usableStr))
+  }
+
+  private lazy val configIndexOpt: Option[(String, Int)] = {
+    argsWithIndex.find(_._1.toLowerCase == "--conf")
+  }
+
+  lazy val baseConfig: Config = configIndexOpt match {
+    case None =>
+      val configPath =
+        datadirPathOpt.getOrElse(AppConfig.DEFAULT_BITCOIN_S_DATADIR)
+      AppConfig
+        .getBaseConfig(configPath)
+        .resolve()
+    case Some((_, configIndex)) =>
+      val str = args(configIndex + 1)
+      val usableStr = str.replace("~", Properties.userHome)
+      val path = Paths.get(usableStr)
+      ConfigFactory
+        .parseFile(path.toFile)
+        .resolve()
+  }
+
+  lazy val configDataDir: Path = Paths.get(
+    baseConfig.getStringOrElse("bitcoin-s.datadir",
```

Comment:
I don't have anything actionable on this for now, but we do want to make sure we are compatible with the resolution hiearchy in the type safe config specification. I listed it out on #2147

Reply:
Actually thinking about this more, is there a reason we don't move 

1. configDataDir
2. datadirPath
3. network 

Could these all be made available via a `BitcoinSAppConfig` in this trait?

Reply:
1) We need all of these so we can properly set the log location.

2) We only use a `BitcoinSAppConfig` in the app server, the oracle server uses a `DLCOracleAppConfig`